Modules/_io/textio.c

```diff
@@ -1992,6 +1992,13 @@ _io_TextIOWrapper_read_impl(textio *self, Py_ssize_t n)
         if (bytes == NULL)
             goto fail;
 
+        if (bytes == Py_None){
+            PyErr_SetString(PyExc_BlockingIOError, "Unexpected None encountered. This may be due to non-blocking I/O "
```

Comment:
Lets keep the error message simple (in both versions), "read returned None".

Reply:
Done.